app/src/main/java/org/desarrolladorslp/teckersapp/ui/messages/PriorityholderFragment.kt

```diff
@@ -0,0 +1,62 @@
+package org.desarrolladorslp.teckersapp.ui.messages
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import org.desarrolladorslp.teckersapp.R
+
+class PriorityholderFragment : Fragment()
+{
+    private lateinit var messageViewModel: MessageViewModel
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var viewAdapter: RecyclerView.Adapter<MessageAdapter.MessageHeaderHolder>
+    private lateinit var viewManager: RecyclerView.LayoutManager
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        messageViewModel =
+            ViewModelProviders.of(this).get(MessageViewModel::class.java)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val root = inflater.inflate(R.layout.content_messages, container, false)
+        viewManager = LinearLayoutManager(context)
+        messageViewModel._inbox.observe(activity as AppCompatActivity, Observer{ inbox ->
+            viewAdapter = MessageAdapter(messageViewModel.getmessages(arguments?.getInt(ARG_SECTION_NUMBER) ?: 1))
```

Comment:
```suggestion
            viewAdapter = MessageAdapter(
                if (arguments?.getInt(ARG_SECTION_NUMBER) == 1) inbox.highPriority
                else inbox.lowPriority
            )
```
This is how you should use LiveData. You already have access to the inbox object here. What would I do better? I would move the HighPriority and LowPriority inbox selection logic to the Inbox data class.